lib/SPIRV/SPIRVToLLVMDbgTran.cpp

```diff
@@ -1267,6 +1267,23 @@ SPIRVToLLVMDbgTran::transTypeTemplateParameter(const SPIRVExtInst *DebugInst) {
                                                              false);
 }
 
+// Workaround for the incorrect enum used for OpTypeTemplateTemplateParameter
+// and OpTypeTemplateParameterPack instructions previously. The W/A added
+// to enforce compatibility with previously generated SPIR-V modules.
+DINode *SPIRVToLLVMDbgTran::transTypeTemplateParameterInst(
+    const SPIRVExtInst *DebugInst) {
+  constexpr SPIRVWord IndexToCheck =
+      SPIRVDebug::Operand::TypeTemplateTemplateParameter::TemplateNameIdx;
+  const SPIRVWordVec &Ops = DebugInst->getArguments();
+  // We can't distinguish these 2 instructions by number of SPIR-V words which
+  // would be better. But we know, that the second parameter of
+  // OpTypeTemplateTemplateParameter is 'Template Name', which for
+  // OpTypeTemplateParameterPack it is 'Source'.
+  if (BM->get<SPIRVValue>(Ops[IndexToCheck])->getOpCode() == OpString)
```

Comment:
Ah yeah, we have a workaround like this locally. However, I was worried that it could be `DebugInfoNone` (I don't find the spec very clear about which operands are allowed to be `DebugInfoNone`)

I came up with a multi-stage guesstimation firstly using the number of operands, which may be more than 10 for only `DebugTypeTemplateParameterPack`, or failing that this operand being `OpString` for `DebugTypeTemplateTemplateParameter`, or this operand being `DebugSource` for `DebugTypeTemplateParameterPack` or if both of those are `DebugInfoNone` then some tie breaker I've forgotten about. It was all a bit complicated.

If this can only truly ever be `OpString` and never `DebugInfoNone` then this looks fine to me.

Reply:
Indeed